### DISCUSSION TITLE
Sending signed transactions require including private keys

### DISCUSSION THREAD

**Discussion by ghost** [near/near-jsonrpc-client-rs#20] [Q&A]
This is how to create signed transactions https://github.com/near/nearcore/blob/master/nearcore/src/runtime/mod.rs#L2245-L2253

The `&signer` object contains public and private key. 

Then used here https://github.com/near/near-jsonrpc-client-rs/blob/master/src/methods.rs#L294-L296

Converts signed transaction to json here https://github.com/near/near-jsonrpc-client-rs/blob/master/src/methods.rs#L312-L314

Sent over network here https://github.com/near/near-jsonrpc-client-rs/blob/master/src/lib.rs#L264

Can we not send signed transactions without including private key?

**Comment by frol** [comment]
To get a singed transaction, you need to sign a hash of a borsh-serialized unsigned `Transaction` object with a private key of the signer/sender account. You don't **include** private key, you just need it to construct the `SignedTransaction`, e.g.:

https://github.com/frol/near-cli/blob/0023ab8d20f7a97fd8e61bd2d0808cb2016f086d/src/construct_transaction_subcommand/transfer_subcommand/mod.rs#L16-L30
